renderer/modules/settings/installations/components/Installations/Installations.contoller.tsx

```diff
@@ -33,38 +35,36 @@ export default class Installations extends Component<
       page: 0,
       rowsPerPage: 5,
     },
+    openConfirmDialog: false,
+    uninstallIdePath: '',
   };
   allInstallations: DevonIdeScript[] = [];
 
   constructor(props: unknown) {
     super(props);
   }
 
+  dialogTitle = 'Uninstalling IDE';
+  dialogContent =
+    'Devonfw Dashboard will no longer keep track of this IDE. Are you sure?';
+
   componentDidMount(): void {
     this.getInstallations();
-    this.downloadCompleteHandler();
   }
 
   componentWillUnmount(): void {
-    global.ipcRenderer.removeAllListeners('get:devonIdeScripts');
-    global.ipcRenderer.removeAllListeners('download completed');
+    global.ipcRenderer.removeAllListeners('get:installedVersions');
   }
 
   getInstallations = (): void => {
-    global.ipcRenderer.send('fetch:devonIdeScripts');
-    global.ipcRenderer.on(
-      'get:devonIdeScripts',
-      (_: IpcRendererEvent, installations: DevonIdeScript) => {
-        this.setState((prev) => ({
-          installations: [...prev.installations, installations],
-          query: '',
-        }));
-        this.allInstallations.push(installations);
-      }
-    );
+    global.ipcRenderer.invoke('get:installedVersions').then((result) => {
+      const installations = [...result];
+      this.setState({ installations });
+      this.allInstallations.push(...installations);
+    });
```

Comment:
The previous implemenentation with an event listener (`on(...)`) was usefull because the changelog required to set a variable (check file ides-installation-status.service.ts, method getDevonfwIDEsStatus)  that takes a bit to be filled so I send partial results to the frontend because otherwise the user would see an empty table for a long time.
You can check it switching to a branch without your new changes.

If you really want to keep the `.invoke` promise approach, at least add a spinner to the table while the ide versions are being loaded so the user knows that it's loading